bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -264,23 +325,28 @@ public String getProductIdentifier(SlingHttpServletRequest request) {
             return identifier;
         }
 
-        ProductUrlFormat.Params productIdentifiers = newProductUrlFormat.parse(request.getRequestPathInfo(),
-            request.getRequestParameterMap());
+        ProductUrlFormat.Params productIdentifiers = getUrlFormatFromContext(request, null, PN_PRODUCT_PAGE_URL_FORMAT,
+            DEFAULT_PRODUCT_URL_FORMATS, systemDefaultProductUrlFormat, productPageUrlFormat, newProductUrlFormat,
+            f -> new ProductPageUrlFormatAdapter(f)).parse(
+                request.getRequestPathInfo(),
+                request.getRequestParameterMap());
 
         // if we get the product sku from URL no extra lookup is needed
         if (StringUtils.isNotEmpty(productIdentifiers.getSku())) {
             identifier = productIdentifiers.getSku();
         } else if (StringUtils.isNotEmpty(productIdentifiers.getUrlKey())) {
-            // lookup internal product identifier (sku) based on URL product identifier (url_key)
+            // lookup internal product identifier (sku) based on URL product identifier
+            // (url_key)
             MagentoGraphqlClient magentoGraphqlClient = request.adaptTo(MagentoGraphqlClient.class);
             if (magentoGraphqlClient != null) {
                 UrlToProductRetriever productRetriever = new UrlToProductRetriever(magentoGraphqlClient);
                 productRetriever.setIdentifier(productIdentifiers.getUrlKey());
                 ProductInterface product = productRetriever.fetchProduct();
                 identifier = product != null ? product.getSku() : null;
             } else {
-                LOGGER.warn("No backend GraphQL client provided, cannot retrieve product identifier for {}", request.getRequestURL()
-                    .toString());
+                LOGGER.warn("No backend GraphQL client provided, cannot retrieve product identifier for {}",
```

Comment:
Please review the formatting changes, there are many new line breaks that are not necessarily needed.